packages/devextreme/testing/testcafe/tests/editors/dateBox/keyboard.ts

```diff
@@ -0,0 +1,698 @@
+
+  await t
+    .click(dateBox.input);
+
+  await t
+    .expect(dateBox.option('opened'))
+    .eql(true);
+
+  await t
+    .pressKey('tab');
+
+  await t
+    .expect(dateBox.getPopup().getNavigatorPrevButton().isFocused)
+    .eql(true);
+
+  await t
+    .pressKey('esc');
+
+  await t
+    .expect(dateBox.option('opened'))
+    .eql(false);
+
+  await t
+    .click(dateBox.input);
+
+  await t
+    .expect(dateBox.option('opened'))
+    .eql(true);
+
+  await t
+    .pressKey('tab')
+    .pressKey('tab');
+
+  await t
+    .expect(dateBox.getPopup().getNavigatorCaption().isFocused)
+    .eql(true);
+
+  await t
+    .pressKey('esc');
+
+  await t
+    .expect(dateBox.option('opened'))
+    .eql(false);
+
+  await t
+    .click(dateBox.input);
+
+  await t
+    .expect(dateBox.option('opened'))
+    .eql(true);
+
+  await t
+    .pressKey('tab')
+    .pressKey('tab')
+    .pressKey('tab');
+
+  await t
+    .expect(dateBox.getPopup().getNavigatorNextButton().isFocused)
+    .eql(true);
+
+  await t
+    .pressKey('esc');
+
+  await t
+    .expect(dateBox.option('opened'))
+    .eql(false);
+
+  await t
+    .click(dateBox.input);
+
+  await t
+    .expect(dateBox.option('opened'))
+    .eql(true);
+
+  await t
+    .pressKey('tab')
+    .pressKey('tab')
+    .pressKey('tab')
+    .pressKey('tab');
+
+  await t
+    .expect(dateBox.getPopup().getViewsWrapper().focused)
+    .eql(true);
+
+  await t
+    .pressKey('esc');
+
+  await t
+    .expect(dateBox.option('opened'))
+    .eql(false);
+
+  await t
+    .click(dateBox.input);
+
+  await t
+    .expect(dateBox.option('opened'))
+    .eql(true);
+
+  await t
+    .pressKey('tab')
+    .pressKey('tab')
+    .pressKey('tab')
+    .pressKey('tab')
+    .pressKey('tab');
+
+  await t
+    .expect(dateBox.getPopup().getTodayButton().isFocused)
+    .eql(true);
+
+  await t
+    .pressKey('esc');
+
+  await t
+    .expect(dateBox.option('opened'))
+    .eql(false);
+
+  await t
+    .click(dateBox.input);
+
+  await t
+    .expect(dateBox.option('opened'))
+    .eql(true);
+
+  await t
+    .pressKey('tab')
+    .pressKey('tab')
+    .pressKey('tab')
+    .pressKey('tab')
+    .pressKey('tab')
+    .pressKey('tab');
+
+  await t
+    .expect(dateBox.getPopup().getCancelButton().isFocused)
+    .eql(true);
+
+  await t
+    .pressKey('esc');
+
+  await t
+    .expect(dateBox.option('opened'))
+    .eql(false);
+
+  await t
+    .click(dateBox.input);
+
+  await t
+    .expect(dateBox.option('opened'))
+    .eql(true);
+
+  await t
+    .pressKey('tab')
+    .pressKey('tab')
+    .pressKey('tab')
+    .pressKey('tab')
+    .pressKey('tab')
+    .pressKey('tab')
+    .pressKey('tab');
+
+  await t
+    .expect(dateBox.getPopup().getApplyButton().isFocused)
+    .eql(true);
+
+  await t
+    .pressKey('esc');
+
+  await t
+    .expect(dateBox.option('opened'))
+    .eql(false);
+}).before(async () => createWidget('dxDateBox', {
+  openOnFieldClick: true,
+  applyValueMode: 'useButtons',
+}));
```

Comment:
This is a really large test. Can we split it?

Reply:
splited on three tests